discovery-infra/tests/test_kube_api.py

```diff
@@ -0,0 +1,292 @@
+import base64
+import errno
+import json
+import logging
+import os
+import socket
+import uuid
+from copy import deepcopy
+from ipaddress import IPv4Interface, IPv6Interface
+
+import magic
+import openshift as oc
+import pytest
+import waiting
+from netaddr import IPNetwork
+from test_infra import consts
+from test_infra.helper_classes.kube_helpers import (
+    InstallStrategy, Proxy, deploy_default_cluster_deployment,
+    deploy_default_infraenv)
+from test_infra.utils import download_iso
+
+from tests.base_test import BaseTest
+from tests.conftest import env_variables
+
+PROXY_PORT = 3129
+
+logger = logging.getLogger(__name__)
+
+
+class TestKubeAPISNO(BaseTest):
+
+    @pytest.mark.kube_api
+    def test_kube_api_sno_happy_flow_create_and_install_cluster(
+        self,
+        _sno_environment_variables,
+        kube_api_context,
+        nodes
+    ):
+        sno_kube_api_test(kube_api_context, nodes)
+
+    @pytest.mark.kube_api_local
+    def test_kube_api_ipv6_sno_happy_flow_create_and_install_cluster(
+        self,
+        _sno_ipv6_environment_variables,
+        _sno_environment_variables,
+        kube_api_context,
+        proxy_server,
+        nodes,
+    ):
+        sno_kube_api_test(
+            kube_api_context,
+            nodes,
+            proxy_server,
+            is_ipv4=False,
+            is_disconnected=False
+        )
+
+    @pytest.fixture(scope='function')
+    def _sno_environment_variables(self):
+        orig = deepcopy(env_variables)
+        env_variables['bootstrap_in_place'] = True
+        env_variables['master_count'] = 1
+        env_variables['num_masters'] = 1
+        env_variables['num_workers'] = 0
+        env_variables['num_nodes'] = 1
```

Comment:
we don't yet have the new mechanism of ``Config`` class?

Reply:
Not yet. This would be removed once https://github.com/openshift/assisted-test-infra/pull/746 is merged.